src/Microsoft.Diagnostics.Monitoring/DiagnosticsMonitor.cs

```diff
@@ -61,19 +56,16 @@ public Task<Stream> ProcessEvents(int processId, TimeSpan duration, Cancellation
                     throw new InvalidOperationException("Failed to start the event pipe session", ex);
                 }
 
-                CancellationTokenSource linkedSource = CancellationTokenSource.CreateLinkedTokenSource(_stopProcessingSource.Token, cancellationToken);
-
                 _currentTask = Task.Run( async () =>
                 {
-                    try
-                    {
-                        await Task.Delay(duration, linkedSource.Token);
-                    }
-                    finally
-                    {
-                        linkedSource.Dispose();
-                        StopSession(session);
-                    }
+                    using var durationSource = new CancellationTokenSource(duration);
+                    using var durationRegistration = durationSource.Token.Register(() => _stopProcessingSource.TrySetResult(null));
```

Comment:
Don't we still want this to be a linked token source? This disregards the original cancellation token.

Reply:
Fixed